localstack/utils/bootstrap.py

```diff
@@ -616,8 +616,9 @@ def configure_container(container: LocalstackContainer):
 
     configure_volume_mounts(container)
 
+    # TODO: no need in podman container (docker runs inside podman without this)
     # mount docker socket
-    container.volumes.append((config.DOCKER_SOCK, config.DOCKER_SOCK))
+    # container.volumes.append((config.DOCKER_SOCK, config.DOCKER_SOCK))
```

Comment:
I was thinking this change is going to break a lot of tests, but in the end, it was the only one that checks `/var/run/docker.sock`:  https://app.circleci.com/pipelines/github/localstack/localstack/3183/workflows/d91ddf3d-7b70-43c3-bb93-66c4584398e5/jobs/13048

I was surprised by this a lot.

@whummer @thrau do we really need to mount thedocker socket nowadays?

Reply:
@naspeh I think what happens currently is that we transparently fall back to the `local` Lambda executor in case Docker is not available from within the container

Under MacOS, if I start up like this:
```
docker run -e LAMBDA_EXECUTOR=docker --rm -it -p 4566:4566 localstack/localstack
```
... then seeing this in the startup logs:
```
2021-12-15T10:47:24.986:WARNING:localstack.services.awslambda.lambda_api: Lambda executor configured as LAMBDA_EXECUTOR=docker but Docker is not accessible. Please make sure to mount the Docker socket /var/run/docker.sock into the container.
```
and finally a Lambda invocation fails with:
```
...
  File "/opt/code/localstack/localstack/utils/docker_utils.py", line 1131, in client
    raise ContainerException("Docker not available")
localstack.utils.docker_utils.ContainerException: Docker not available
...
  File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 648, in run_lambda_executor
    raise InvocationException(
localstack.services.awslambda.lambda_executors.InvocationException: Lambda process returned with error. Result: . Output:
```

Reply:
@whummer Thank you for the highlight

I'm testing `podman` container with `LAMBDA_EXECUTOR=docker` and I can make it working yet, `podman` has a different IP for
https://github.com/localstack/localstack/blob/ff3817a3be403019487b38d05f043da57b67f385/localstack/config.py#L621-L629

I'm looking at how I can solve this.

Reply:
I was able to use `LAMBDA_EXECUTOR=docker` with `podman` but I have to configure a bunch of stuff

I installed https://github.com/containers/dnsname and then modified `/etc/cni/net.d/87-podman.conflist` with the section from https://github.com/containers/dnsname/blob/main/README_PODMAN.md
```
     {
        "type": "dnsname",
        "domainName": "dns.podman",
        "capabilities": {
           "aliases": true
        }
     }
```
then I ran
```
podman system migrate
```
and was able to run with this
```
sudo sh -c 'DEBUG=1 MAIN_CONTAINER_NAME=localstack DOCKER_BRIDGE_IP=$MAIN_CONTAINER_NAME LAMBDA_EXECUTOR=docker localstack start'
```
I redefined `MAIN_CONTAINER_NAME` as well because `botocore` refuses `localstack_main` domain name https://github.com/boto/botocore/blob/14b13fb86982f62385c794ee9a8ac35907cbfb26/botocore/utils.py#L1061-L1064

----

I have to figure out the simplest solution for this.

Reply:
Fixed `get_main_endpoint_from_container` here https://github.com/localstack/localstack/pull/5124/commits/b07f2e3fb9a095850b8eafb7092127c900aa37c5